### PR TITLE
Port attribute helpers to CE3

### PR DESCRIPTION
**Pull request by catostrophe**
+ merge other updates

### CODE REVIEW

File: modules/core/src/main/scala/io/janstenpickle/trace4cats/attributes/HostAttributes.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import cats.syntax.functor._
+import io.janstenpickle.trace4cats.model.AttributeValue.StringValue
+import io.janstenpickle.trace4cats.model.{AttributeValue, SemanticAttributeKeys}
+
+import java.net.InetAddress
+
+object HostAttributes {
+  def apply[F[_]: Sync]: F[Map[String, AttributeValue]] =
+    for {
+      inetAddress <- Sync[F].blocking(InetAddress.getLocalHost)
```
Comment (catostrophe) [inline modules/core/src/main/scala/io/janstenpickle/trace4cats/attributes/HostAttributes.scala:14]: @janstenpickle in CE3 these should be blocking calls

File: modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import io.janstenpickle.trace4cats.http4s.common.Request_
+import io.janstenpickle.trace4cats.model.AttributeValue.{LongValue, StringValue}
+import io.janstenpickle.trace4cats.model.{AttributeValue, SemanticAttributeKeys}
+import org.http4s.Uri
+
+object Http4sClientRequest {
+  def toAttributes(req: Request_): Map[String, AttributeValue] =
+    req.uri.host.map { host =>
+      val key = host match {
```
Comment (catostrophe) [inline modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala:11]: @janstenpickle please look. I changed it slightly. And I think we should change it in CE2.

File: modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sHeaders.scala [hunk 3]
```diff
@@ -18,18 +20,26 @@ object Http4sHeaders {
...
-    List[(String, AttributeValue)](
-      "http.method" -> req.method.name,
-      "http.url" -> req.uri.path.toString
-    ) ++ headerFields(req.headers, "req", dropHeadersWhen)
+    List[(String, AttributeValue)](httpMethod -> req.method.name, httpUrl -> req.uri.path.toString) ++ headerFields(
+      req.headers,
+      "req",
+      dropHeadersWhen
+    ) ++ req.uri.host.map { host =>
+      val key = host match {
```
Comment (catostrophe) [inline modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sHeaders.scala:28]: @janstenpickle same here

File: modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala (hunk 2 above)
Comment (janstenpickle) [reply modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala:11]: Oh nice change, yeah let's backport to CE2

**Pull request merged**